File: source/includes/fts-tutorial/sort/numbers-query.go
package main

import (
	"context"
	"fmt"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func main() {
	// connect to your Atlas cluster
	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI("<connection-string>"))
	if err != nil {
		panic(err)
	}
	defer client.Disconnect(context.TODO())

	// set namespace
	collection := client.Database("sample_mflix").Collection("movies")

	// define pipeline stages
	searchStage := bson.D{{"$search", bson.D{
	  {"index", "sort-tutorial"},
	  {"range", bson.D{
		{"path", "awards.wins"},
		{"gte", 10},
	  }},
	  {"sort", bson.D{{"awards.wins", -1}}},
	}}}
	limitStage := bson.D{{"$limit", 5}}
	projectStage := bson.D{{"$project", bson.D{{"title", 1}, {"awards.wins", 1}, {"_id", 0}}}}

	// run pipeline
	cursor, err := collection.Aggregate(context.TODO(), mongo.Pipeline{searchStage, limitStage, projectStage})
	if err != nil {
		panic(err)
	}

	// print results
	var results []bson.D
	if err = cursor.All(context.TODO(), &results); err != nil {
		panic(err)
	}
	for _, result := range results {
		fmt.Println(result)
	}
}

File: source/includes/fts-tutorial/sort/numbers-query.py
import pymongo

# connect to your Atlas cluster
client = pymongo.MongoClient('<connection-string>')

# define pipeline
pipeline = [
  {
    '$search': {
      'index': 'sort-tutorial',
      'range': {
        'path': 'awards.wins',
        'gte': 10
      },
      'sort': {
        'awards.wins': -1
      }
    }
  }, {
    '$limit': 5
  }, {
    '$project': {'_id': 0, 'title': 1, 'awards.wins': 1
    }
  }
]

# run pipeline
result = client['sample_mflix']['movies'].aggregate(pipeline)

# print results
for i in result:
    print(i)
